### DISCUSSION TITLE
Filter usability - operators case sensitive setting

### DISCUSSION THREAD

**Discussion by jdjoshuadavison** [directus/directus#19721] [Feature Requests]
### The overarching problem
Users find the search and filters difficult to use.
This is one of a number of requests to help solve this problem.

### Case sensitive
The list of operators is very long. To reduce the length of this list, we could remove the case sensitive variants and add a case sensitive toggle to the filter rule row.

### Motivation

Users find the filters difficult to use.

### Detailed Design

A case sensitive toggle could potentially look like this:
Disabled:
![filter-ui-case-sensitive-toggle-off](https://github.com/directus/directus/assets/34007858/0788597a-f828-46be-9b9e-312b9fbc48c4)

Enabled:
![filter-ui-case-sensitive-toggle-on](https://github.com/directus/directus/assets/34007858/fb061dd6-e5ac-41a7-9d4e-9fd3d6be0eaa)

(a tooltip on hover would be helpful)

### Must Have:
- Case sensitivity toggle

### Drawbacks

Ability to change case sensitivity may be less obvious than before (unless a better design can be found).

### Alternatives

- The case sensitive option could be removed altogether, and filters could always be case insensitive 😲 - it seems unlikely you'd want to do this, but I'm just throwing it out there, how often do people need/want their filters to be case sensitive? 

- Another alternative - **or additional** - solution could be to format the list of operators. See: #19720

### Adoption Strategy

Assuming case sensitive and case insensitive variations are simply stored as 2 different operators, you could continue to store the data that way, and simply handle it at a UI level. This would ensure no breaking changes and remove any need for migration.

If you want to change it to store the operator and the case sensitive setting separately, then existing filters would break, and a migration would be required.

### Unresolved Questions

- Is the potentially reduced visibility of the case sensitive option worth it to shorten the list of operators?
- How often do people need/want their filters to be case sensitive?

**Comment by ImaCrea** [comment]
Very great idea here! I've seen many users struggling with it and your ux suggestion would clearly be a win

**Comment by krysialynne** [comment]
+1 to this; I really like the case-sensitive toggle UI you've shown.

Another alternative possibility or addition to this: allow users to set default filter operators for a specific field in a collection, or globally for all fields of a specific type. As of 10.9.2, if you select a string type field in the filter UI, the default filter operator that populates is "Contains", which is case-sensitive. For instance, if an admin wants all "string" type fields to default to case-insensitive search behavior, they could set that for an entire collection, or per-field. I'm getting ready to roll out Directus to our content managers and I'm anticipating the default case-sensitive "Contains" filter will cause confusion.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
